include/storage/shared_memory.hpp

```diff
@@ -34,10 +34,10 @@ namespace storage
 
 struct OSRMLockFile
 {
-    boost::filesystem::path operator()()
+    template <typename IdentifierT> boost::filesystem::path operator()(const IdentifierT &id)
```

Comment:
Oof, I was hoping that maybe we could get around to not templating this. I skimmed the calls a bit and it seems like this templated type **is** necessary. Just commenting on here in case you have an gedankenblitz (= an idea within a time span of < 1s) on how to avoid this template.

Reply:
Meh, I could just make it support `operator()(int id)`, but I figured why not make it generally re-usable.

Whatever `id` is needs to be supported by `std::to_string(id)` lower down.